types/jquery/jquery-tests.ts

```diff
@@ -4342,24 +4446,366 @@ function JQueryStatic() {
     }
 
     function when() {
-        const t = $.ajax() as JQuery.jqXHR<string>;
-        const u = $.ajax() as JQuery.jqXHR<number>;
-        const v = $.ajax() as JQuery.jqXHR<boolean>;
+        function Promise3() {
+            const t = $.ajax() as JQuery.jqXHR<string>;
+            const u = $.ajax() as JQuery.jqXHR<number>;
+            const v = $.ajax() as JQuery.jqXHR<boolean>;
+
+            // 3 parameters
+            {
+                const w = $.when(t, u, v);
+
+                w.then((a, b, c) => {
```

Comment:
That's not valid in JQuery 3.

JQuery 3 `.then` and `.catch` always only pass exactly one argument to their callback, like the Promises/A+ spec. `.done` and `.fail` should be used instead if multiple arguments are required.

I shipped this bug into production earlier this year :cry: